### PR TITLE
Add auto-correct to rescue modifier

### PR DESCRIPTION
**Pull request by rrosenblum**

### CODE REVIEW

File: lib/rubocop/cop/mixin/autocorrect_alignment.rb [hunk 1]
```diff
@@ -10,6 +10,14 @@ def configured_indentation_width
         config.for_cop('IndentationWidth')['Width']
       end
 
+      def indentation(node)
+        offset(node) + (' ' * configured_indentation_width)
```
Comment (bbatsov) [inline lib/rubocop/cop/mixin/autocorrect_alignment.rb]: I'm guessing where allocating this ` ` a lot. He should probably make it a frozen constant.

**Comment by rrosenblum** [comment]
> I'm guessing where allocating this `' '` a lot. He should probably make it a frozen constant. 

I have extracted the `' '` to a constant. 

I am working on another commit that will extract commonly used variables to constants. I have been using [memory_profiler](https://github.com/SamSaffron/memory_profiler/tree/6ff1bb359e5dc0ff15603cb71687f6ee6d75cfc6) to identify areas that can be cleaned up. So far, I have reduced memory allocation by RuboCop by about 8%.

**Comment by bbatsov** [comment]
> I am working on another commit that will extract commonly used variables to constants. I have been using memory_profiler to identify areas that can be cleaned up. So far, I have reduced memory allocation by RuboCop by about 8%.

Excellent! We should probably spend some time in the future on performance optimizations as well. People with huge projects often complain that running RuboCop takes quite a while.

**Comment by rrosenblum** [comment]
I am all for making RuboCop more performant. #2047 address auto-correct, but it does not address over all performance.

**Pull request merged**